src/commands/migrate.js

```diff
@@ -8,9 +15,45 @@ const migrate = (context, fromVersion, toVersion, optionalPercent = '100%') => {
     return Promise.resolve();
   }
   optionalPercent = parseInt(optionalPercent, 10);
+
+  let app;
+
   return utils
     .getLinkedApp()
-    .then(app => {
+    .then(linkedApp => {
+      app = linkedApp;
+      if (
+        optionalPercent === 100 &&
```

Comment:
is there any value in setting this threshold a little lower? presumably moving 90% of users to a version means you want that version to be public too. I could even see a case for going lower, maybe 75%.

Reply:
It's hard to define such a threshold. You can always ask why this value and why not that value. So I think it's easier for us to assume when a dev does a partial migration, they're testing. In addition, for private apps, the backend does implicit promotion (such as moving ZTs to the new version) only if it's a 100% migration. To make it symmetry on the public apps, shouldn't we only suggest for promotion only when it's 100% migration?

Reply:
Yeah, that's fair! It's a pattern devs have some issues with, but maybe the consistency is worthwhile